docs/src/test/scala/docs/http/scaladsl/HttpServerExampleSpec.scala

```diff
@@ -394,7 +394,7 @@ class HttpServerExampleSpec extends AnyWordSpec with Matchers
         concat(
           pathEnd {
             concat(
-              (put | parameter("method".requiredValue("put"))) {
+              (put | parameter("method".requiredValue("put")).tmap(_ => ())) {
```

Comment:
I think the DSL becomes more regular when you always get a value for a parameter definition. The `tmap` is more of a distraction.
```suggestion
              (put | parameter("method".requiredValue("put"))) { _ => 
```
In the case of `requiredValue` it could even be that fixed value, but that would break the API too much.

Reply:
In that case, the `tmap` (or some equivalent) is required, though, to make the `|`-operator work. That only works when both sides extract the same number of objects.

Reply:
We could rather say, to show this particular example is a distraction for the example, since it is such a rare case.